### PR TITLE
Add Kerl

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -31,7 +31,9 @@ x25519 = [ "x25519-dalek" ]
 random = [ "getrandom" ]
 aes = [ "aes-gcm", "cipher" ]
 blake2b = [ "blake2", "digest" ]
-curl-p = [ "bee-ternary" ]
+sponge = []
+curl-p = [ "bee-ternary", "sponge" ]
+kerl = [ "bee-ternary", "sponge", "byteorder", "lazy_static", "tiny-keccak" ]
```
Comment (nothingismagick) [inline Cargo.toml]: didn't we want to group these into a feature known as "deprecated_do_not_use" or something?

Comment (thibault-martinez) [reply Cargo.toml]: Renamed

File: Cargo.toml [hunk 2]
```diff
@@ -31,7 +31,9 @@ x25519 = [ "x25519-dalek" ]
 random = [ "getrandom" ]
 aes = [ "aes-gcm", "cipher" ]
 blake2b = [ "blake2", "digest" ]
-curl-p = [ "bee-ternary" ]
+sponge = []
+curl-p = [ "bee-ternary", "sponge" ]
+kerl_deprecated_do_not_use = [ "bee-ternary", "sponge", "byteorder", "lazy_static", "tiny-keccak" ]
```
Comment (l1h3r) [inline Cargo.toml:36]: 👍

**Review by nothingismagick — APPROVED** [review]

**Review by l1h3r — APPROVED** [review]
🤯

File: .changes/kerl.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+"cli.rs": minor
```
Comment (nothingismagick) [inline .changes/kerl.md]:
```suggestion
"iota-crypto": minor
```

**Pull request merged**